sapi/fpm/fpm/fpm_conf.c

```diff
@@ -859,7 +859,7 @@ static int fpm_conf_process_all_pools(void)
 
 		/* alert if user is not set; only if we are root and fpm is not running with --allow-to-run-as-root */
 		if (!wp->config->user && !geteuid() && !fpm_globals.run_as_root) {
-			zlog(ZLOG_ALERT, "[pool %s] user has not been defined", wp->config->name);
+			zlog(ZLOG_ALERT, "[pool %s] FPM is running as root and the user directive has not been specified, refusing to run for security reasons. Specify either a pool-specific user directive or pass the --allow-to-run-as-root flag.", wp->config->name);
```

Comment:
Seems a bit long. I don't think we should make error messages equal to documentation. I would make it shorter just to give a hint what to search for. Something like

```suggestion
			zlog(ZLOG_ALERT, "[pool %s] user directive has not been specified when running as a root without --allow-to-run-as-root", wp->config->name);
```

It would be great to also cover this by test - there is FPM\Tester::skipIfNotRoot(); for root tests...